### PR TITLE
Update properties of capability

### PR DESCRIPTION
**Pull request by dab246**
### Resolved

- [x] https://github.com/linagora/tmail-flutter/issues/1645

### CODE REVIEW

File: test/jmap/core/session/session_test.dart [hunk 1]
```diff
@@ -383,6 +390,150 @@ void main() {
 
       expect(parsedSession, equals(expectedSession));
     });
+
+    test('get should parsing correctly session with some capabilities miss property', () {
```
Comment (hoangdat) [inline test/jmap/core/session/session_test.dart:394]: IMO, you should create a test case for Cyrus session also

File: lib/jmap/core/capability/submission_capability.dart [hunk 2]
```diff
@@ -1,19 +1,18 @@
...
 
-@UnsignedIntConverter()
-@JsonSerializable()
+@UnsignedIntNullableConverter()
+@JsonSerializable(explicitToJson: true, includeIfNull: false)
 class SubmissionCapability extends CapabilityProperties with EquatableMixin {
-  final UnsignedInt maxDelayedSend;
-  final Set<String> submissionExtensions;
+  final UnsignedInt? maxDelayedSend;
+  final Map<String, List<String>>? submissionExtensions;
```
Comment (hoangdat) [inline lib/jmap/core/capability/submission_capability.dart:13]: any conflict with the original JMAP?

Comment (hoangdat) [reply lib/jmap/core/capability/submission_capability.dart:13]: please create the test case for it also

Comment (dab246) [reply lib/jmap/core/capability/submission_capability.dart:13]: According to the JMAP documentation, then we need to change the variable so it is correct. It is also true for CYRUS's json

Screenshot 2023-03-24 at 18 03 38

**Review by hoangdat — APPROVED** [review]

**Pull request merged**